shared/safecast/src/lib.rs

```diff
@@ -355,9 +355,11 @@ pub trait SafeCast: ByteSafe {
         <Self>::bytesafe();
 
         /* Uninitialized is safe here as we will fill in all of the bytes */
-        let mut ret: T = unsafe { core::mem::uninitialized() };
-        self.cast_copy_into(&mut ret);
-        ret
+        let mut ret = core::mem::MaybeUninit::<T>::uninit();
+        unsafe {
+            self.cast_copy_into(&mut *ret.as_mut_ptr());
```

Comment:
This may be incorrect since we're dereferncing uninitialized data, but requires changing `cast_copy_into` argument from `&mut T` to `*mut T`, which doesn't allow working with slices.
The ptr alternative would look like this:
```
    fn cast_copy_into<T: ByteSafe + ?Sized>(&self, dest: *mut T)
...
           self.cast_copy_into(ret.as_mut_ptr());
```
And a change in documentation.

Reply:
Hard to say. Not toooo worried about it, but the whole cast copy stuff is kinda on the fence anyways.